app/src/org/odk/collect/android/logic/ArchivedFormManagement.java

```diff
@@ -0,0 +1,102 @@
+package org.odk.collect.android.logic;
+
+import org.commcare.android.database.SqlStorage;
+import org.commcare.android.database.user.models.FormRecord;
+import org.commcare.android.javarosa.AndroidLogger;
+import org.commcare.android.tasks.FormRecordCleanupTask;
+import org.commcare.dalvik.application.CommCareApp;
+import org.commcare.dalvik.application.CommCareApplication;
+import org.javarosa.core.services.Logger;
+import org.joda.time.DateTime;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Vector;
+
+/**
+ * Determines when saved forms should be purged.
+ *
+ * @author Phillip Mates (pmates@dimagi.com).
+ */
+public class ArchivedFormManagement {
+
+    /**
+     * Purge saved forms from device that have surpassed the validity date set
+     * by the app
+     *
+     * @param app Used to get the saved form validity date property.
+     */
+    public static void performArchivedFormPurge(CommCareApp app) {
+        int daysSavedFormIsValidFor = getArchivedFormsValidityInDays(app);
+        if (daysSavedFormIsValidFor == -1) {
+            return;
+        }
+
+        DateTime lastValidDate = getLastValidArchivedFormDate(daysSavedFormIsValidFor);
+
+        Vector<Integer> toPurge = getSavedFormsToPurge(lastValidDate);
+
+        for (int formRecord : toPurge) {
+            FormRecordCleanupTask.wipeRecord(CommCareApplication._(), formRecord);
+        }
+    }
+
+    /**
+     * Read the validity range for archived forms out of app preferences.
+     *
+     * @return how long archived forms should kept on the phone. -1 if saved
+     * forms should be kept indefinitely
+     */
+    public static int getArchivedFormsValidityInDays(CommCareApp app) {
+        int daysForReview = -1;
+        String daysToPurge =
+                app.getAppPreferences().getString("cc-days-form-retain", "-1");
```

Comment:
I generally put keys like this into a static class-level variable, although I know this is only being used once so not nearly as important. Just looked funny to me but might just be force of my own habit, so up to you.